content/articles/2017-07-ggplot2-tidy-evaluation.Rmarkdown

```diff
@@ -0,0 +1,150 @@
+---
+title: Tidy evaluation in ggplot2
+date: '2018-07-24'
+slug: ggplot2-tidy-evaluation
+author: Mara Averick
+categories: [package]
+description: >
+  Using tidy evaluation in ggplot2 3.0.0.
+photo:
+  url: https://unsplash.com/photos/8KfCR12oeUM
+  author: Christopher Burns
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>", fig.retina = 2, dpi = 300)
+library(tidyverse)
+```
+<html>
+<style>
+h3 code {
+    font-size: 1em;
+    
+}
+</style>
+</html>
+## Tidy evaluation
+
+One of the biggest changes in [ggplot2](https://ggplot2.tidyverse.org/) 3.0.0 is support for [tidy evaluation](https://adv-r.hadley.nz/evaluation.html#tidy-evaluation), making it more programmable, and more consistent with the rest of the tidyverse. Though this introduces some breaking changes, we believe it to be worthwhile in the interest of improving future code. Here, we outline some of the more prominent of those changes. For our complete list of changes, including symptomatic error messages, please see the [Breaking changes](https://ggplot2.tidyverse.org/news/index.html#breaking-changes) section of the [release notes](https://ggplot2.tidyverse.org/news/index.html#ggplot2-3-0-0).
+
+### `aes()` contains quosures
+
+The primary developer-facing change is that `aes()` now contains [quosures](https://adv-r.hadley.nz/evaluation.html#quosures) (expression + environment pairs), rather than symbols. As a result, you'll need to take a different approach to extracting the information you need. Below, we use a "quote-and-unquote" pattern for the aesthetic parameter `x_var`.
```

Comment:
I think this paragraph and example can be moved to the bottom of the post, and replaced with something like this:

## Developer facing changes

If you're building an extension package on top of ggplot2, you need to be aware that the introduction of tidy evaluation fundamentally changes the data structures that `aes()` uses. In brief:

```R
mapping <- aes(mpg, colour = "smoothed")

# Variables are now stored as quosures
mapping$x

# Constants (atomic vectors of length 1), remain
# as is
mapping$colour
```